crates/compiler/gen_llvm/src/llvm/build.rs

```diff
@@ -2964,136 +2964,61 @@ fn list_literal<'a, 'ctx>(
     let list_length = elems.len();
     let list_length_intval = env.ptr_int().const_int(list_length as _, false);
 
-    // TODO re-enable, currently causes morphic segfaults because it tries to update
-    // constants in-place...
-    // if element_type.is_int_type() {
```

Comment:
you can't just ignore this comment :) the issue is real, the generated code will try to in-place modify the list which is now stored in read-only memory. I think the proper fix for now is to store all bytes in a global, and then use a memcpy instead of a normal loop to allocate the list on the heap when it is instantiated.

Reply:
Is that the issue that we're running into CI? I deleted the comment because I figured that was independent to the implementation in that branch, not the one in this PR.

Reply:
I believe that that is the issue yes. Some in-place mutation is occuring on bytes that are in the readonly section of the binary.